numpy/_array_api/_creation_functions.py

```diff
@@ -0,0 +1,177 @@
+from __future__ import annotations
+
+from ._types import (Optional, SupportsDLPack, SupportsBufferProtocol, Tuple,
+                     Union, array, device, dtype)
+from ._dtypes import _all_dtypes
+
+import numpy as np
+
+def asarray(obj: Union[float, NestedSequence[bool|int|float], SupportsDLPack, SupportsBufferProtocol], /, *, dtype: Optional[dtype] = None, device: Optional[device] = None, copy: Optional[bool] = None) -> array:
```

Comment:
The `NestedSequence` import seems to be missing.

Reply:
Also the type of `obj` will, in practice, be `Any` as long as `SupportsDLPack` and `SupportsBufferProtocol` are effectively untyped.

Reply:
Lastly, note that `np.ndarray` is not a subtype of `npt.NestedSequence`, as the former does not fully implement all `collections.abc.Sequence`-based methods (while type-checking at least, during runtime you could cheat and just manually register the class).